### DISCUSSION TITLE
Return a component in the render method.

### DISCUSSION THREAD

**Discussion by timvandijck** [livewire/livewire#6802] [3. Feature ideas]
I found myself in the scenario where I have a relatively simple overview with a possibility to add a new item through a form. I want to replace the overview with the form when in "add"-mode. 

I found the ability of inline components to be already helpful and my code looks sort of like this:

```php
public function render()
{
    if ($this->showForm) {
        return <<<'HTML'
            <div>
                <livewire:create-form />
            </div>
        HTML;
    }

    return view('livewire.overview');
}
```

Now it would be even more cool if you could just return the component instead, something like this:

```php
public function render()
{
    if ($this->showForm) {
        return CreateForm::class;
    }

    return view('livewire.overview');
}
```

**Comment by lrljoe** [comment]
Use a boolean and toggle whether you want to display overview or form in your blade?

You can use the Form approach to standardise your forms.

Changing how Laravel itself handles rendering would seem a bit odd, if you need a full page component, then use one.

Otherwise, I'd recommend either using traits, or customising what is displayed on page based on user interaction with components.

For example, I return a full page "home" component, this returns a view which has
a table subcomponent for the "view all"/"overview" page

and clicking add new, or edit item opens a modal to do the editing (anothee component)

clicking view returns the "view item" subcomponent

The only reason I have these split into separate components is use of 3rd party packages, you could do it all in one with traits etc should you so wish, and it'd probably be more efficient!
